core/src/ldml/ldml_processor.cpp

```diff
@@ -225,77 +225,86 @@ ldml_processor::process_event(
           state->actions().push_backspace(KM_KBP_BT_UNKNOWN);
         } else {
           state->actions().push_backspace(KM_KBP_BT_CHAR, last_char);
+          state->context().pop_back();
         }
-        state->context().pop_back();
       }
       break;
     default:
-      // from kmx_processor.cpp
-      // Construct a context buffer from the items up until the last KM_KBP_CT_MARKER marker
-      ldml::string_list ctxt;
-      auto cp = state->context();
-      // We're only interested in as much of the context as is a KM_KBP_CT_CHAR.
-      // This will stop at the KM_KBP_CT_MARKER type.
-      uint8_t last_type = KM_KBP_BT_UNKNOWN;
-      for (auto c = cp.rbegin(); c != cp.rend(); c++) {
-        last_type = c->type;
-        if (last_type != KM_KBP_BT_CHAR) {
-          break;
+      {
+        // adapted from kmx_processor.cpp
+
+        /** a copy of the current/changed context, for transform use */
+        ldml::string_list ctxt;
+
+        // Construct a context buffer of all the KM_KBP_BT_CHAR items
+        // Extract the context into 'ctxt' for transforms to process
+        if (!!transforms) {
+          // if no transfors, no reason to do this extraction
+          auto &cp = state->context();
+          // We're only interested in as much of the context as is a KM_KBP_BT_CHAR.
+          uint8_t last_type = KM_KBP_BT_UNKNOWN;
+          for (auto c = cp.rbegin(); c != cp.rend(); c++) {
+            last_type = c->type;
+            if (last_type != KM_KBP_BT_CHAR) {
```

Comment:
Does this cope with markers interleaved in the context?

Reply:
It just stops when it gets to a non char (such as a marker). will need to revisit w/ markers